semgrep/semgrep/autofix.py

```diff
@@ -35,6 +53,38 @@ def _modify_file(rule_match: RuleMatch, fix: str) -> None:
     p.write_text(contents_after_fix_str)
 
 
+def _parse_regex_fix(sed_string: str) -> Tuple[str, str]:
+    """
+    Return the second and third elements of a sed-like string:
+    E.g., s/one/two/g returns (one, two)
+    """
+    splitstr = sed_string.split("/")  # Do it this way to satisfy mypy
+    return splitstr[1], splitstr[2]
+
+
+def _regex_replace(rule_match: RuleMatch, from_str: str, to_str: str) -> None:
+    """
+    Use a regular expression to autofix.
+    Replaces from_str to to_str.
```

Comment:
Add comment that it replaces the first instance of from_str to to_str

Reply:
Based on my read of the docs, it replaces all occurrences as is:  https://docs.python.org/3/library/re.html#re.sub

```python
>>> import re
>>> replaceme = "XaaaaXXaaXXXaaXaaaaaaXX"
>>> re.sub("X", "E", replaceme)
'EaaaaEEaaEEEaaEaaaaaaEE'
>>> 
```

Do you have a preferred behavior? Is it more intuitive to replace everything or just the first occurrence?

(I basically just lifted the `sed` syntax but I always use /g ... and it isn't really `sed` anyway so I'm not sure which is more familiar)

Reply:
Oh i must have misread the docs. Can you add comment that it replaces all occurrences. Alternatively you can use the prescence of /g to make it all behavior but default to one replacement if g isnt there?

Reply:
Yup. I realized after this discussion that a better solution would be to just parse the /g or /N flag so I just added that in.